source/melpomene/src/main.rs

```diff
@@ -69,27 +73,79 @@ fn kernel_entry() {
     {
         let mut guard = k.heap().lock().unwrap();
 
-        // First let's make a dummy driver just to make sure some stuff happens
-        let dummy_chan = KChannel::new(&mut guard, 16);
-
-        k.register_driver(DriverHandle {
-            kind: DriverKind::Todo,
-            queue: dummy_chan.clone(),
-        })
-        .map_err(drop)
-        .unwrap();
+        let send_channel: KChannel<BBQBidiHandle>;
 
-        let dummy_fut = async move {
-            let _ = dummy_chan;
+        {
+            // First let's make a dummy driver just to make sure some stuff happens
+            let dummy_chan = KChannel::new(&mut guard, 16);
+            send_channel = dummy_chan.clone();
 
-            loop {
+            let dummy_fut = async move {
                 Sleepy::new(Duration::from_secs(1)).await;
-                tracing::warn!("Dummy tick...");
-            }
-        };
-        let dummy_task = k.new_task(dummy_fut);
-        let boxed_dummy = guard.alloc_box(dummy_task).map_err(drop).unwrap();
-        k.spawn_allocated(boxed_dummy);
+                let (a_ring, b_ring) = new_bidi_channel(k.heap(), 32, 16).await;
+                dummy_chan
+                    .enqueue_async(b_ring)
+                    .await
+                    .map_err(drop)
+                    .unwrap();
+
+                let mut i = 0u32;
+                loop {
+                    tracing::info!("Driver A: Writing...");
+                    let mut wgr = a_ring.send_grant_exact(8).await;
+                    wgr.iter_mut().for_each(|b| *b = (i % 255) as u8);
+                    i = i.wrapping_add(1);
+                    let len = wgr.len();
+                    wgr.commit(len);
+
+                    tracing::info!("Driver A: Sleeping...");
+                    Sleepy::new(Duration::from_secs(1)).await;
+
+                    tracing::warn!("Driver A: Reading...");
+                    let rgr = a_ring.read_grant().await;
+                    tracing::warn!(
+                        buf = ?rgr.deref(),
+                        "Driver A: Got data"
+                    );
+                    let len = rgr.len();
+                    rgr.release(len);
+                }
+            };
```

Comment:
consider sticking a span on these?
```suggestion
            }.instrument(tracing::info_span!("Driver A");
```
(of course, we'd have to add `use tracing::Instrument;` somewhere)